File: app/src/main/java/com/linwoain/storytelling/adapter/BookAdapter.java
package com.linwoain.storytelling.adapter;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import com.koushikdutta.ion.Ion;
import com.linwoain.library.LViewHelper;
import com.linwoain.library.LinAdapter;
import com.linwoain.storytelling.R;
import com.linwoain.storytelling.bean.BookInfo;
import com.linwoain.util.LLStringTools;
import java.util.List;

/**
 * create by wuxuejian on 2016/6/3
 */
public class BookAdapter extends LinAdapter<BookInfo> {
  /**
   * LinAdapter通用的构造方法
   *
   * @param context 传入的上下文
   * @param beans 要显示的数据源封装好的列表
   */
  public BookAdapter(Activity context, List<BookInfo> beans) {
    super(context, beans);
  }

  protected View LgetView(int position, View convertView, ViewGroup parent) {
    if (convertView==null) {
      convertView = LViewHelper.getView(R.layout.item_main, context);
    }
    ImageView icon = ViewHolders.get(convertView, R.id.icon);
    //ImageButton start = ViewHolders.get(convertView, R.id.start);
    BookInfo bookInfo = beans.get(position);
    Ion.with(icon)
        .error(R.drawable.listen)
        .load(bookInfo.getCoverImage());
    ((TextView) ViewHolders.get(convertView, R.id.title)).setText(bookInfo.getName());
    String title = bookInfo.getTitle();
    if (!LLStringTools.isEmpty(title)) {
      ((TextView) ViewHolders.get(convertView, R.id.status)).setText(
          String.format("已经听到：%s", title));
    }

    return convertView;
  }
}

File: app/src/main/java/com/linwoain/storytelling/adapter/BookAdapter.kt
package com.linwoain.storytelling.adapter

import android.app.Activity
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.koushikdutta.ion.Ion
import com.linwoain.library.LViewHelper
import com.linwoain.library.LinAdapter
import com.linwoain.storytelling.R
import com.linwoain.storytelling.bean.Novel
import com.linwoain.util.LLStringTools

/**
 * create by linwoain on 2016/6/3
 *  LinAdapter通用的构造方法
 * @param context 传入的上下文
 * @param beans 要显示的数据源封装好的列表
 */
class BookAdapter(context: Activity, beans: List<Novel>) : LinAdapter<Novel>(context, beans) {

    override fun LgetView(position: Int, view: View?, parent: ViewGroup): View {
        var convertView = view
        if (convertView == null) {
            convertView = LViewHelper.getView(R.layout.item_main, context)
        }
        val novel = beans[position]
        val icon = LinAdapter.ViewHolders.get<ImageView>(convertView, R.id.icon)
        Ion.with(icon).placeholder(R.drawable.listen).error(R.drawable.listen).load(novel.icon)

        ViewHolders.get<TextView>(convertView, R.id.title).text = novel.name
        if (!LLStringTools.isEmpty(novel.title)) {
            ViewHolders.get<TextView>(convertView, R.id.status).text = String.format("已经听到：%s", novel.title)
        }


        return convertView!!
    }
}
